specs/core/0_beacon-chain.md

```diff
@@ -1330,90 +1324,42 @@ def process_deposit(state: BeaconState,
     return index
 ```
 
-### Routine for updating validator status
+### Routines for updating validator status
 
-```python
-def update_validator_status(state: BeaconState,
-                            index: int,
-                            new_status: int) -> None:
-    """
-    Update the validator status with the given ``index`` to ``new_status``.
-    Handle other general accounting related to this status update.
-    Note that this function mutates ``state``.
-    """
-    if new_status == ACTIVE:
-        activate_validator(state, index)
-    if new_status == ACTIVE_PENDING_EXIT:
-        initiate_validator_exit(state, index)
-    if new_status in [EXITED_WITH_PENALTY, EXITED_WITHOUT_PENALTY]:
-        exit_validator(state, index, new_status)
-```
-
-The following are helpers and should only be called via `update_validator_status`:
+_Note that all functions in this section mutate `state`_.
 
 ```python
-def activate_validator(state: BeaconState,
-                       index: int) -> None:
-    """
-    Activate the validator with the given ``index``.
-    Note that this function mutates ``state``.
-    """
+def activate_validator(state: BeaconState, index: int, immediate: bool) -> None:
```

Comment:
Consider renaming `immediate` to `genesis` for clarity.

(Side question: could we always do `validator.activation_slot = state.slot + ENTRY_EXIT_DELAY`, including at genesis?)

Reply:
Renamed to `genesis`.

> (Side question: could we always do validator.activation_slot = state.slot + ENTRY_EXIT_DELAY, including at genesis?)

I suppose we could, but not sure what the value is...